core/src/main/kotlin/org/evomaster/core/problem/externalservice/httpws/param/HttpWsResponseParam.kt

```diff
@@ -54,23 +59,43 @@ class HttpWsResponseParam (
     /**
      * @return HTTP status code
      */
-    fun getHttpStatusCode() : Int = status.values[status.index]
+    fun getHttpStatusCode(): Int {
+        return status.values[status.index]
+    }
+
+    /**
+     * @return HTTP 200 if the [responseBody] has value.
+     * @return HTTP 204 if the body is blank
+     *
+     *  All 1xx (Informational), 204 (No Content), and 304 (Not Modified) responses do not include content.
+     *  https://www.rfc-editor.org/rfc/rfc9110.html#section-6.4.1-8
+     */
+    fun getResponseBodyBasedOnStatus(): String {
+        val statusCode = getHttpStatusCode()
+
+        if (arrayOf(204, 304).contains(statusCode)) {
```

Comment:
wouldn't hurt to check if not within 100 and 199

Reply:
added